include/boost/url/rfc/impl/absolute_uri_rule.ipp

```diff
@@ -16,19 +16,44 @@
 namespace boost {
 namespace urls {
 
-void
-absolute_uri_rule::
+auto
+absolute_uri_rule_t::
 parse(
     char const*& it,
-    char const* const end,
-    error_code& ec,
-    absolute_uri_rule& t) noexcept
+    char const* const end
+        ) const noexcept ->
+    result<value_type>
 {
-    grammar::parse(
-        it, end, ec,
-        t.scheme_part,
-        t.hier_part,
-        t.query_part);
+    value_type t;
+
+    // scheme
```

Comment:
Yes, this is what I was saying before, that these changes add technical debt, because I just refactored everything as quickly as possible, without an eye towards improving it. After we merge this I am going to do a pass through the implementation of these rules to see how they can be tidied up. And some of them will be made private.